f5_openstack_agent/lbaasv2/drivers/bigip/agent_manager_lite.py

```diff
@@ -438,10 +443,176 @@ def scrub_dead_agents_in_env_and_group(self, context):
+            else:
+                member_status = constants_v2.F5_ONLINE
+        else:
+            LOG.warning("Unexpected status %s and session %s",
+                        member.state, member.session)
+        return member_status
+
+    @log_helpers.log_method_call
+    def update_member_status_per_pool(self, pool_id, tenant_id,
+                                      hostnames=list()):
+        """update the members' status in one pool """
+        all_members = list()
+        # further discussions are needed
+        try:
+            for bigip in self.lbdriver.get_active_bigips():
+                if bigip.hostname in hostnames:
+                    pool_name = self.conf.environment_prefix + '_' + pool_id
+                    partition = self.conf.environment_prefix + '_' + tenant_id
+                    pool = resource_helper.BigIPResourceHelper(
+                        resource_helper.ResourceType.pool).load(
+                            bigip, name=pool_name, partition=partition)
+                    members = pool.members_s.get_collection()
+                    # figure out the members in this pool and send
+                    # the members and their statuses to driver in batch
+                    for member in members:
+                        # check if the member's description
+                        # follows the <prefix>-<id> rule
+                        if hasattr(member, 'description'):
+                            description = member.description
+                            if not description:
+                                LOG.debug("member description is empty.")
+                                continue
+
+                            member_parts = description.split('_', 1)
+                            if len(member_parts) == 2:
+                                # member_prefix = member_parts[0].strip()
+                                member_id = member_parts[1].strip()
+                                if not self.is_valid_uuid(str(member_id)):
+                                    LOG.debug("member_id %s isn't uuid format."
+                                              % member_id)
+                                    continue
+                            else:
+                                LOG.debug("member description is wrong %s."
+                                          % description)
+                                continue
+
+                        else:
+                            LOG.debug("membe description is None.")
+                            continue
+
+                        member_info = {}
+                        member_info['id'] = member_id
+                        member_info['state'] = \
+                            self.calculate_member_status(member)
+                        LOG.debug("append member %s with statue %s",
+                                  member_id,
+                                  member_info['state'])
+                        all_members.append(member_info)
```

Comment:
If you have multiple bigips, will you need to merge your member list fetched from every bigips?

Reply:
This logic is consistent with the old way that is just retrieve the members from the active bigip.

Reply:
change the logic from get_active_bigips to get_active_bigip which is consistent with the previous member update logic.